hnn_core/gui.py

```diff
@@ -0,0 +1,474 @@
+                 tstart=tstart, tstart_std=tstart_std,
+                 burst_rate=burst_rate, burst_std=burst_std,
+                 repeats=repeats, seedcore=seedcore,
+                 tstop=tstop)
+    drive.update(widgets_dict)
+    return drive, drive_box
+
+
+def _get_poisson_widget(name, tstop_widget, layout, style, location):
+    tstart = FloatText(value=0.0, description='Start time (s)',
+                       layout=layout, style=style)
+    tstop = BoundedFloatText(value=tstop_widget.value,
+                             max=tstop_widget.value,
+                             description='Stop time (s)',
+                             layout=layout, style=style)
+    seedcore = IntText(value=14, description='Seed',
+                       layout=layout, style=style)
+    location = RadioButtons(options=['proximal', 'distal'])
+
+    cell_types = ['L5_pyramidal', 'L2_pyramidal', 'L5_basket',
+                  'L2_basket']
+    rate_constant = dict()
+    for cell_type in cell_types:
+        rate_constant[f'{cell_type}'] = FloatText(
+            value=8.5, description=f'{cell_type}:',
+            layout=layout, style=style)
+
+    widgets_list, widgets_dict = _get_cell_specific_widgets(layout, style, location)
+    widgets_dict.update({'rate_constant': rate_constant})
+    widgets_list.extend([HTML(value="<b>Rate constants</b>")] +
+                        list(widgets_dict['rate_constant'].values()))
+
+    drive_box = VBox([tstart, tstop, seedcore] + widgets_list)
+    drive = dict(type='Poisson', name=name, tstart=tstart,
+                 tstop=tstop, rate_constant=rate_constant,
+                 seedcore=seedcore,
+                  )
+    drive.update(widgets_dict)
+    return drive, drive_box
+
+
+def _get_evoked_widget(name, layout, style, location):
+    kwargs = dict(layout=layout, style=style)
+    mu = FloatText(value=0, description='Mean time:', **kwargs)
+    sigma = FloatText(value=1, description='Std dev time:',
+                      **kwargs)
+    numspikes = IntText(value=1, description='No. Spikes:',
+                        **kwargs)
+    seedcore = IntText(value=14, description='Seed: ', **kwargs)
+
+    widgets_list, widgets_dict = _get_cell_specific_widgets(layout, style, location)
+
+    drive_box = VBox([mu, sigma, numspikes, seedcore] +
+                     widgets_list)
+    drive = dict(type='Evoked', name=name,
+                 mu=mu, sigma=sigma, numspikes=numspikes,
+                 seedcore=seedcore,
+                 sync_within_trial=False)
+    drive.update(widgets_dict)
+    return drive, drive_box
+
+
+def add_drive_widget(drive_type, drive_boxes, drive_widgets,
+                     drives_out, tstop_widget, location):
+    """Add a widget for a new drive."""
+    layout = Layout(width='270px', height='auto')
+    style = {'description_width': '150px'}
+    drives_out.clear_output()
+    with drives_out:
+        name = drive_type + str(len(drive_boxes))
```

Comment:
Good opportunity for a unit test here, checking if the adding two of the same drive produces the current name